src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -457,4 +475,23 @@ object Task {
       else None
     }
   }
+
+  implicit class TaskStatusComparison(val task: Task) extends AnyVal {
+    def isReserved: Boolean = task.status.taskStatus == MarathonTaskStatus.Reserved
+    def isCreated: Boolean = task.status.taskStatus == MarathonTaskStatus.Created
+    def isError: Boolean = task.status.taskStatus == MarathonTaskStatus.Error
+    def isFailed: Boolean = task.status.taskStatus == MarathonTaskStatus.Failed
+    def isFinished: Boolean = task.status.taskStatus == MarathonTaskStatus.Finished
+    def isKilled: Boolean = task.status.taskStatus == MarathonTaskStatus.Killed
+    def isKilling: Boolean = task.status.taskStatus == MarathonTaskStatus.Killing
+    def isRunning: Boolean = task.status.taskStatus == MarathonTaskStatus.Running
+    def isStaging: Boolean = task.status.taskStatus == MarathonTaskStatus.Staging
+    def isStarting: Boolean = task.status.taskStatus == MarathonTaskStatus.Starting
+    def isUnreachable: Boolean = task.status.taskStatus == MarathonTaskStatus.Unreachable
+    def isGone: Boolean = task.status.taskStatus == MarathonTaskStatus.Gone
+    def isUnknown: Boolean = task.status.taskStatus == MarathonTaskStatus.Unknown
+    def isDropped: Boolean = task.status.taskStatus == MarathonTaskStatus.Dropped
+
+    def isReachable: Boolean = isStaging || isStarting || isRunning || isKilling
```

Comment:
Please add a scaladoc here, as it is not clear from the name what is included. From the perspective of `!unreachable`, the given set makes sense. Yet, from the perspective of _can we reach this task_, this would probably only be `isRunning || is Killing` (we can't perform a HealthCheck against a staging or starting task).